Character.cpp

```diff
@@ -98,20 +97,44 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        // a
+        if( hitPoints < *initialHitPoints ) 
+        {
+            hitPoints = *initialHitPoints;
+        } 
+
+        if( armor < *initialArmorLevel )
+        {
+            armor = *initialArmorLevel;
+        }
+
+        if( attackDamage < *initialAttackDamage )
+        {
+            attackDamage = *initialAttackDamage;
+        }
+
+        // b
+        hitPoints += hitPoints * 0.1;
+        armor += armor * 0.1;
+        attackDamage += attackDamage * 0.1;
+
+        // c
+        initialHitPoints.reset( new int(hitPoints) );
+        initialArmorLevel.reset( new int(armor) );
+        initialAttackDamage.reset( new int(attackDamage) );
```

Comment:
1) you're doing the same comparison and assignment to all 3 pairs of variables.   D.R.Y.   write a helper function. 
2) don't reset the `initial___` variables.  dereference and assign.
- [ ] revise